File: Homework/Homework05ConditionalStatements/Triangle/Triangle.cs
using System;

class Triangle
{
    static void Main()
    {
        double aX = double.Parse(Console.ReadLine());
        double aY = double.Parse(Console.ReadLine());
        double bX = double.Parse(Console.ReadLine());
        double bY = double.Parse(Console.ReadLine());
        double cX = double.Parse(Console.ReadLine());
        double cY = double.Parse(Console.ReadLine());

        double a = Math.Sqrt(Math.Pow((bX - aX), 2) + Math.Pow((bY - aY), 2));
        double b = Math.Sqrt(Math.Pow((cX - bX), 2) + Math.Pow((cY - bY), 2));
        double c = Math.Sqrt(Math.Pow((aX - cX), 2) + Math.Pow((aY - cY), 2));

        double p = (a + b + c) / 2;

        if ((a + b) > c && (b + c) > a && (a + c) > b)
        {
            Console.WriteLine("Yes");
            double area = Math.Sqrt(p * (p - a) * (p - b) * (p - c));
            Console.WriteLine("{0:F2}", area);
        }
        else
        {
            Console.WriteLine("No");
            Console.WriteLine("{0:F2}", a);
        }
    }
}

File: Homework/08-OOP-in-Java-Homework/01Geometry/src/models/planeShapes/Triangle.java
package models.planeShapes;

import models.PlaneShape;
import models.Vertex2D;

public class Triangle extends PlaneShape {
    public Triangle(Vertex2D a, Vertex2D b, Vertex2D c) {
        super(new Vertex2D[]{a, b, c});
    }

    @Override
    public double getPerimeter() {
        double perimeter = this.getSides()[0] + this.getSides()[1] + this.getSides()[2];

        return perimeter;
    }

    @Override
    public String getShapeName() {
        return "Triangle";
    }

    // Heron's Formula
    @Override
    public double getArea() {
        double area = Math.sqrt(
                (this.getPerimeter() / 2) *
                ((this.getPerimeter() / 2 - this.getSides()[0]) *
                (this.getPerimeter() / 2 - this.getSides()[1]) *
                (this.getPerimeter() / 2 - this.getSides()[2])));

        return area;
    }

    @Override
    public String toString() {
        StringBuilder triangleInfo = new StringBuilder();
        triangleInfo.append(String.format("%s\r\n", this.getShapeName()));
        triangleInfo.append(String.format(
        "point coordinates: a %s, b %s, c %s\r\n",
        getVertices2D()[0],
        getVertices2D()[1],
        getVertices2D()[2]));
        triangleInfo.append(String.format("area = %.2f\r\n", this.getArea()));
        triangleInfo.append(String.format("perimeter = %.2f", this.getPerimeter()));

        return triangleInfo.toString();
    }

    private double[] getSides() {
        double firstSide = getDistanceBetweenTwoPoints(getVertices2D()[1], getVertices2D()[2]);
        double secondSide = getDistanceBetweenTwoPoints(getVertices2D()[2], getVertices2D()[0]);
        double thirdSide = getDistanceBetweenTwoPoints(getVertices2D()[0], getVertices2D()[1]);
        double[] sides = {firstSide, secondSide, thirdSide};

        return sides;
    }

    private double getDistanceBetweenTwoPoints(Vertex2D firstPoint, Vertex2D secondPoint) {
        double deltaX = firstPoint.getX() - secondPoint.getX();
        double deltaY = firstPoint.getY() - secondPoint.getY();

        double distance = Math.sqrt(
                Math.pow(deltaX, 2) +
                Math.pow(deltaY, 2));

        return distance;
    }
}